File: ABC/ABC266/C.cpp
#include<iostream>
using namespace std;

struct Point {
  int x, y;
};

bool f(Point&a, Point&b, Point&c, Point&d) {
  int dx = a.x-b.x;
  int dy = a.y-b.y;
  int vc = dx * (c.y - a.y) - dy * (c.x - a.x);
  int vd = dx * (d.y - a.y) - dy * (d.x - a.x);
  if(vc == 0 || vd == 0) return false;
  if(vc * vd > 0) return true;
  return false;
}

int main() {
  Point a, b, c, d;
  cin >> a.x >> a.y >> b.x >> b.y >> c.x >> c.y >> d.x >> d.y;
  bool ans = true;
  ans &= f(a, b, c, d);
  ans &= f(b, c, d, a);
  ans &= f(c, d, a, b);
  ans &= f(d, a, b, c);
  if(ans) cout << "Yes" << endl;
  else cout << "No" << endl;
  return 0;
}

File: ABC/ABC266/C.rs
use proconio::input;
use std::ops;

#[derive(Debug)]
struct Point {
  x: i32,
  y: i32
}

fn outer(a: &Point, b: &Point, o: &Point) -> bool {
  let oa = Point {x: a.x - o.x, y: a.y - o.y,};
  let ob = Point {x: b.x - o.x, y: b.y - o.y,};
  if oa.x * ob.y - oa.y * ob.x <= 0 {return false;}
  else {return true;}
}

fn main() {
  let mut v: Vec<Point> = Vec::new();
  for i in 0..4 {
    input! {x: i32, y: i32}
    v.push(Point{x: x, y: y});
  }

  let mut ans: bool = true;
  ans &= outer(&v[2], &v[0], &v[1]);
  ans &= outer(&v[3], &v[1], &v[2]);
  ans &= outer(&v[0], &v[2], &v[3]);
  ans &= outer(&v[1], &v[3], &v[0]);
  if ans {
    println!("Yes");
  } else {
    println!("No");
  }

}